src/Microsoft.SqlTools.ServiceLayer/TableDesigner/TableDesignerService.cs

```diff
@@ -161,13 +161,17 @@ private Task HandleGenerateScriptRequest(TableInfo tableInfo, RequestContext<str
             });
         }
 
-        private Task HandleGeneratePreviewReportRequest(TableInfo tableInfo, RequestContext<string> requestContext)
+        private Task HandleGeneratePreviewReportRequest(TableInfo tableInfo, RequestContext<GeneratePreviewReportResult> requestContext)
         {
-            return this.HandleRequest<string>(requestContext, async () =>
+            return this.HandleRequest<GeneratePreviewReportResult>(requestContext, async () =>
             {
                 var table = this.GetTableDesigner(tableInfo);
                 var report = table.GenerateReport();
-                await requestContext.SendResult(report);
+                var generatePreviewReportResult = new GeneratePreviewReportResult();
+                // TODO - set report type by caohai
+                generatePreviewReportResult.Report = report;
+                generatePreviewReportResult.MimeType = "text/markdown";
```

Comment:
Shouldn't this be `text/plain`?

Reply:
default is markdown
this will be changed by @caohai anyway

Reply:
The report is being generated as markdown? I thought you had to hardcode the markdown string in your other PR because it wasn't being generated as markdown yet.

Reply:
Hai's working on doing that. He'll pick up from this PR.

Reply:
It would be better to have this reflect the actual type that's being sent. He can update it as part of his PR then...or are you implying that he's going to be adding stuff to this PR?